pkg/build/builder/docker.go

```diff
@@ -297,6 +297,11 @@ func (d *DockerBuilder) dockerBuild(dir string, tag string, secrets []buildapi.S
 		return err
 	}
 
+	network, resolvConfHostPath, err := getContainerNetworkMode()
```

Comment:
It would be a lot better if this is an interface passed into docker builder, and none of this code had any idea of what this stuff was.

Reply:
what exactly are you proposing be an interface?  A ContainerNetworkModeGetter interface?